src/web/transactions.rs

```diff
@@ -23,62 +25,45 @@ pub enum TransactionDTO {
     Rejected(RejectedTransactionDTO),
 }
 
-impl TryFrom<TransactionValue> for TransactionDTO {
+impl TryFrom<TransactionQueryResult> for TransactionDTO {
     type Error = color_eyre::Report;
 
-    fn try_from(tx: TransactionValue) -> Result<Self> {
-        match tx {
-            TransactionValue::Transaction(versioned) => {
-                let tx = versioned.into_v1();
-                let base = TransactionBase::new(tx.hash(), tx.payload, tx.signatures)
-                    .wrap_err("Failed to make TransactionBase")?;
-                Ok(Self::Committed(CommittedTransactionDTO { base }))
-            }
-            TransactionValue::RejectedTransaction(versioned) => {
-                let tx = versioned.into_v1();
-                let hash = tx.hash();
-                let RejectedTransaction {
-                    payload,
-                    signatures,
-                    rejection_reason,
-                } = tx;
-                let base = TransactionBase::new(hash, payload, signatures)
-                    .wrap_err("Failed to make TransactionBase")?;
-                Ok(Self::Rejected(RejectedTransactionDTO {
-                    base,
-                    rejection_reason: rejection_reason.into(),
-                }))
-            }
+    fn try_from(tx_result: TransactionQueryResult) -> Result<Self> {
+        let tx_value = tx_result.transaction();
+        let TransactionValue { tx, error } = tx_value;
+        let hash = tx.hash();
+        let base = TransactionBase::new(hash, tx.payload().clone(), tx.signatures().clone())
+            .wrap_err("Failed to make TransactionBase")?;
+
+        match error {
+            Some(rejection_reason) => Ok(Self::Rejected(RejectedTransactionDTO {
+                base,
+                rejection_reason: (*rejection_reason).clone().into(),
+            })),
+            None => Ok(Self::Committed(CommittedTransactionDTO { base })),
         }
     }
 }
 
 #[derive(Serialize)]
 struct TransactionBase {
-    hash: SerScaleHex<HashOf<SignedTransaction>>,
-    block_hash: SerScaleHex<Hash>,
-    payload: TransactionPayloadDTO,
-    signatures: BTreeSet<SignatureDTO>,
+    hash: HashOf<VersionedSignedTransaction>,
+    payload: TransactionPayload,
+    signatures: Vec<SignaturesOf<TransactionPayload>>,
```

Comment:
1. Why have you removed `SerScaleHex` wrap?
2. Why have you removed `block_hash` field from the public API?